### ISSUE TITLE
JSON handling for parameterized topologies

### ISSUE THREAD

**Issue by CalCraven** [mosdef-hub/gmso#795]
Seems like there's an issue saving the JSON state of a topology, where the parameters aren't stored at all. Maybe this is a good place to discuss if that's a feature we would want to have.

```python
import gmso
from gmso.parameterization import apply
top = gmso.Topology.load("ethane.json")
assert not top.is_typed()
ff = gmso.ForceField("ethane.xml")
ptop = apply(top, ff, identify_connections=True)
assert ptop.is_typed()
ptop.save("ethane-typed.json")
newtop = gmso.Topology("ethane-typed.json")
print(newtop.is_typed("topology"))
```

output
```
False
```

**Comment by daico007** [comment]
I think this is already supported, likely need to change the default options and clarify the documentation for this method (https://github.com/mosdef-hub/gmso/blob/main/gmso/formats/json.py#L24-L42)

**Comment by daico007** [comment]
Yes, totally support changing the default to `True` (for saving out a typed structure)

**Comment by daico007** [comment]
This has been updated in the latest commit in https://github.com/mosdef-hub/gmso/pull/794

**Comment by daico007** [comment]
Addressed in #794
